src/Monolog/Logger.php

```diff
@@ -311,6 +308,11 @@ public function addRecord(int $level, string $message, array $context = []): boo
             $record = call_user_func($processor, $record);
         }
 
+        reset($this->handlers);
+        while ($key !== key($this->handlers)) {
+            next($this->handlers);
```

Comment:
as you dont do anything in this look, can't you just use `end` ?
http://php.net/manual/en/function.end.php